### PR TITLE
Set the actor ID as the assigned task ID for direct actor workers

### PR DESCRIPTION
**Pull request by stephanie-wang**
## Why are these changes needed?

The worker's assigned task ID is used by the raylet to track which plasma objects each local worker needs. This was set to NIL for all direct actors, which was causing test failures like [this](https://api.travis-ci.com/v3/job/261224391/log.txt). It could also cause a bug in fetching objects from remote nodes, since all direct actors would share the same task ID.

This fixes the bug by assigning direct actors' their actor ID as the current task ID. It's a bit hacky, but we should be able to remove most of this code once we move actor creation tasks to use direct calls.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -1081,12 +1081,15 @@ void NodeManager::HandleWorkerAvailable(
 void NodeManager::HandleWorkerAvailable(const std::shared_ptr<Worker> &worker) {
   RAY_CHECK(worker);
   // If the worker was assigned a task, mark it as finished.
+  bool worker_available = true;
```
Comment (ericl) [inline src/ray/raylet/node_manager.cc]: Consider bool is_actor

**Review by ericl — APPROVED** [review]
Looks fine as an interim fix. Just one suggestion on naming.

File: src/ray/raylet/node_manager.cc (hunk 1 above)
Comment (stephanie-wang) [reply src/ray/raylet/node_manager.cc]: Hmm yeah I'll call it is_direct_actor since we don't actually do that for raylet actors.

Comment (stephanie-wang) [reply src/ray/raylet/node_manager.cc]: Or I guess is_not_direct_actor...

**Review by zhijunfu — APPROVED** [review]
LGTM.

btw: The actor creation task should probably still go to raylet then to the gcs service per today's discussion, but that's outside of the scope of this PR and can be discussed separately later.

**Pull request merged**